core/src/main/java/com/graphhopper/routing/ch/PrepareCHEdgeIterator.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing.ch;
+
+import com.graphhopper.routing.profiles.BooleanEncodedValue;
+import com.graphhopper.routing.weighting.Weighting;
+import com.graphhopper.util.CHEdgeExplorer;
+import com.graphhopper.util.CHEdgeIterator;
+import com.graphhopper.util.EdgeIterator;
+
+public class PrepareCHEdgeIterator implements PrepareCHEdgeExplorer {
+    private final CHEdgeExplorer edgeExplorer;
+    private final Weighting weighting;
+    private final BooleanEncodedValue accessEnc;
+    private CHEdgeIterator chIterator;
+
+    public PrepareCHEdgeIterator(CHEdgeExplorer edgeExplorer, Weighting weighting) {
+        this.edgeExplorer = edgeExplorer;
+        this.weighting = weighting;
+        this.accessEnc = weighting.getFlagEncoder().getAccessEnc();
+    }
+
+    @Override
+    public PrepareCHEdgeIterator setBaseNode(int node) {
+        chIterator = edgeExplorer.setBaseNode(node);
+        return this;
+    }
+
+    public boolean next() {
+        return iter().next();
+    }
+
+    public int getEdge() {
+        return iter().getEdge();
+    }
+
+    public int getBaseNode() {
+        return iter().getBaseNode();
+    }
+
+    public int getAdjNode() {
+        return iter().getAdjNode();
+    }
+
+    public boolean isForward() {
+        return iter().get(accessEnc);
+    }
+
+    public boolean isBackward() {
+        return iter().getReverse(accessEnc);
+    }
+
+    public int getOrigEdgeFirst() {
+        return iter().getOrigEdgeFirst();
+    }
+
+    public int getOrigEdgeLast() {
+        return iter().getOrigEdgeLast();
+    }
+
+    public boolean isShortcut() {
+        return iter().isShortcut();
+    }
+
+    public double getWeight(boolean reverse) {
+        if (isShortcut()) {
+            return iter().getWeight();
+        } else {
+            return weighting.calcWeight(iter(), reverse, EdgeIterator.NO_EDGE);
```

Comment:
Do you intend to use PrepareCHEdgeIterator for edge-based traversal too - then we would need a separate parameter?

Reply:
Its already being used for edge-based CH preparation. This method only calculates the edge-weight (use `NO_EDGE` for the `nextOrPrev` parameter). So its more like we could remove the parameter rather than we need another one.